src/DynamicData/Kernel/Optional.cs

```diff
@@ -185,13 +186,17 @@ public static class Optional
     /// </summary>
     /// <typeparam name="T">The type of the item.</typeparam>
     /// <returns>The optional value.</returns>
-    public static Optional<T> None<T>() => Optional<T>.None;
+    public static Optional<T> None<T>()
+        where T : notnull
+        => Optional<T>.None;
 
     /// <summary>
     /// Wraps the specified value in an Optional container.
     /// </summary>
     /// <typeparam name="T">The type of the item.</typeparam>
     /// <param name="value">The value.</param>
     /// <returns>The optional value.</returns>
-    public static Optional<T> Some<T>(T value) => new(value);
+    public static Optional<T> Some<T>(T? value)
```

Comment:
Value could be null as the constructor of optional directly casts a null to a None<T>.

Reply:
Doh. Ignore me as I just noticed the T?